tensorflow-framework/src/main/java/org/tensorflow/framework/losses/CategoricalHinge.java

```diff
@@ -0,0 +1,91 @@
+package org.tensorflow.framework.losses;
+
+import org.tensorflow.Operand;
+import org.tensorflow.framework.losses.impl.LossesImpl;
+import org.tensorflow.op.Ops;
+import org.tensorflow.types.family.TNumber;
+
+/**
+ * Computes the categorical hinge loss between labels and predictions.
```

Comment:
Should we follow the Python in documenting that labels are expected to be `0` or `1`?

Reply:
Yes, The Python `CategporicalHinge` class does not mention that at all, but it is mentioned in the `loss_hinge`.
method.

I have added an entry to the class JavaDoc and to the `Losses.categoricalHinge` method.

Reply:
Actually the values can be` [-1, 0, 1]`.` [0,1]` is converted to `[-1,1`]. I have added a value check to make sure the values are wholly contained in the allowed values set ` [-1, 0, 1]`. This will either throw `TFInvalidArgumentException` if run in Graph mode via a control dependency, and throw `IllegalArgumentException` if created in Eager mode with the `call` method.

Reply:
Cool -- Resolved.

Reply:
What does it do if there are all three of `[-1 0 1]` present? That's probably an invalid input, does it throw?